### PR TITLE
Clientside: Block low lvl for high lvl spells - Fixes #12904

### LINKED ISSUE #12904: Skills you can't use at your current Level should not subtract Mana
As reported by @citrusella :

"Casting a skill you don't have access to yet (i.e. a level 13 skill like Tools of the Trade when you're only level 12) deducts mana even as it tells you you need to be level [x] to cast it."_

The Mana is returned after a full sync has happened, but the expected behaviour is that it should never be deducted.

The screenshot below was taken immediately after I clicked on the level 13 skill.

image

### PR DESCRIPTION
**Pull request by Helcostr**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #12904

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

- Block low lvl players from visually using high lvl spells (resulting in visual mana)
- Add i18n for gold (an edit from b4) as well as aforementioned lvl
- Remove trailing whitespace while I was digging around in i18n

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 10cbe941-0e10-4763-9f4f-7a3ae5bb2198

### CODE REVIEW

**Comment by Helcostr** [comment]
I was not able to run `npm test` as requested by the guideline for push request: eslint wasn't set up properly.

```
> habitica@4.185.0 test D:\GitHub\habitica
> npm run lint && gulp test && gulp apidoc


> habitica@4.185.0 lint D:\GitHub\habitica
> eslint --ext .js --fix . && cd website/client && npm run lint
Oops! Something went wrong! :(

ESLint: 7.14.0

ESLint couldn't find the config "habitrpg/lib/node" to extend from. Please check that the name of the config is correct.

The config "habitrpg/lib/node" was referenced from the config file in "D:\GitHub\habitica\.eslintrc.js".

If you still have problems, please stop by https://eslint.org/chat/help to chat with the team.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! habitica@4.185.0 lint: `eslint --ext .js --fix . && cd website/client && npm run lint`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the habitica@4.185.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\helco\AppData\Roaming\npm-cache\_logs\2021-03-27T14_03_21_307Z-debug.log
npm ERR! Test failed.  See above for more details.
```

This was probably because I was deploying to a docker, and editing on the parent.

But my changes were so small, it shouldn't fail... I hope...

**Comment by Helcostr** [comment]
Agh whoops... didn't mean to delete branch.

File: website/client/src/mixins/spells.js [hunk 1]
```diff
@@ -34,10 +34,11 @@ export default {
         return null;
       }
 
+      if (this.user.stats.lvl < spell.lvl) return this.text(this.$t('notEnoughLvl'));
```
Comment (SabreCat) [inline website/client/src/mixins/spells.js]: It's not even necessary to throw an error here, methinks--the not-yet-unlocked skills are quite visually distinct from the unlocked ones, so swallowing the click silently is still intuitive.

I could see hiding the details popover for skills that aren't unlocked yet, but that needn't be in scope for this PR.

**Review by SabreCat — CHANGES_REQUESTED** [review]
One comment where I think we could simplify a tiny bit, but this is working well!

File: website/client/src/mixins/spells.js (hunk 1 above)
Comment (Helcostr) [reply website/client/src/mixins/spells.js]: How do you want me to handle the i18n code `return this.text(this.$t('notEnoughLvl'));`? Would u like me to leave it in as a comment for future reference, or do u want me to cut it out?

Comment (SabreCat) [reply website/client/src/mixins/spells.js]: OK to remove it!

Comment (Helcostr) [reply website/client/src/mixins/spells.js]: Got it. Issues is now resolved.

**Comment by Helcostr** [comment]
I am not sure why the API check is failing... does someone know?

**Review by SabreCat — APPROVED** [review]
I like the way this feels now. Thanks!

**Comment by SabreCat** [comment]
Noted toward Tier 3! Much appreciated!

**Pull request merged**